[NXP] Integrating code updates for RW61X and K32W platforms

dinabenamar opened the pull request:
- Integrating code updates for K32W0 & K32W1 platforms
- Integrating code updates for RW61X & SDK 2.13.3 support
- Enabling CI workflows for RW61X platform for all-clusters, thermostat, and laundry-washer applications

marius-alex-tache committed:
[NXP][scripts][common] Add checkout_submodules NXP target

All submodules that don't have a platforms field will be selected
when the script is run. Additionally, if --platform nxp is used,
submodules that have nxp under their platform field will also be
checked out.

Signed-off-by: marius-alex-tache <marius.tache@nxp.com>

marius-alex-tache committed:
[NXP][scripts][common] Fix lint errors

Signed-off-by: marius-alex-tache <marius.tache@nxp.com>

chapongatien committed:
[NXP][doc][rw61x] Adding information to run bootstrap script + submodule update script

+ Updating gn information to build Matter target that will support BR

Signed-off-by: Gatien Chapon <gatien.chapon@nxp.com>

dinabenamar committed:
[NXP][CI][rw61x] Enabling CI workflow for RW61x examples (all-clusters-app, thermostat, laundry-washer)

marian-chereji-nxp committed:
[NXP][platform] Fix string keys saving failure

The "NXPConfig" class member function which is supposed to save the
string keys to persistent storage was, in fact, invoking the
"NvSaveOnIdle" function using the INTEGER keys RAM buffer descriptor,
instead of using the STRING keys RAM buffer descriptor. Most likely
a typical copy-paste error.

Signed-off-by: Marian Chereji <marian.chereji@nxp.com>

dinabenamar committed:
[NXP][platform][common] Adding platform common code

dinabenamar committed:
[NXP][platform][k32w] Adding platform code changes for k32w

dinabenamar committed:
[NXP][platform][rw61x] Adding platform code changes for rw61x

dinabenamar committed:
[NXP][tools][common] Adding nxp tools updates

dinabenamar committed:
[NXP][examples][rw61x] Adding examples updates

dinabenamar committed:
[NXP][examples][k32w] Adding examples updates

dinabenamar committed:
[NXP][doc][common] Adding documentation updates

dinabenamar committed:
[NXP][examples][common] Adding examples common updates

marius-alex-tache committed:
[NXP][platform][k32w0] Move subscription and groups KVS keys to dedicated storages

To avoid a large KVS RAM buffer and memory allocation issues when running TC-RR-1.1,
move some specific keys in dedicated RAM storage instaces:
* subscription keys will be stored in sSubscriptionStorage.
* groups keys will be stored in sGroupsStorage. This RAM storage will use the extended
search feature, because its size will be greater than current backend region size (2K).

The key/value storage will be selected in each API based on some
criteria: if the key is related to either subscription/groups, then
both the key and the value will be stored in the same storage. Otherwise,
the default keys and values storages are used.

Signed-off-by: marius-alex-tache <marius.tache@nxp.com>

marius-alex-tache committed:
[NXP][examples][k32w0] Update default args.gni for reference apps

Signed-off-by: marius-alex-tache <marius.tache@nxp.com>

marius-alex-tache committed:
[NXP][examples][k32w1] Update default args.gni for reference apps

Signed-off-by: marius-alex-tache <marius.tache@nxp.com>

Martin-NXP committed:
[NXP][examples][common] Fix missing QR code log

marius-alex-tache committed:
[NXP][platform][rw61x] Add define for encrypted factory data

CONFIG_CHIP_ENCRYPTED_FACTORY_DATA is defined by default when factory data
is enabled. The common code should leverage this flag to make some actions
when implicit encrypted factory data is enabled.

Signed-off-by: marius-alex-tache <marius.tache@nxp.com>

marius-alex-tache committed:
[NXP][examples][common][factory-data] Enclose encryption code

Encrypted factory data related code should be enclosed based on the
CONFIG_CHIP_ENCRYPTED_FACTORY_DATA flag.

Signed-off-by: marius-alex-tache <marius.tache@nxp.com>

marius-alex-tache committed:
[NXP][examples][common] Add a way to include a specific OTAImageProcessor header

CONFIG_CHIP_OTA_IMAGE_PROCESSOR_HEADER should specify an OTA image processor implementation
header. It can be a custom one, such as the header for OTA multi-image implementation.
This will ease the transition of current reference apps that are not using the processor
from the NXP common area.

Signed-off-by: marius-alex-tache <marius.tache@nxp.com>

marius-alex-tache committed:
[NXP][examples][common] Enclose binding usage based on EMBER_AF_PLUGIN_BINDING

EMBER_AF_PLUGIN_BINDING is defined when the server side of the Binding cluster is used.
The common app task code should initialize Binding handlers only if this flag is defined.

Signed-off-by: marius-alex-tache <marius.tache@nxp.com>

marius-alex-tache committed:
[NXP][platform][k32w0] Add conversion for subscription and groups keys

When CONFIG_CHIP_K32W0_KVS_MOVE_KEYS_TO_SPECIFIC_STORAGE is set to 1,
the application can opt to move some keys from current KVS storage
to a dedicated storage, based on the key name.

Currently, the mechanism supports moving keys and values for:
* subscriptions
* groups

Note: the system is meant to ensure backwards compatibility. This should only
be used once, but, if this feature is still enabled after the first conversion,
the corresponding keys will not be found in KVS default storage. This does
not affect the functionality, but it introduces a delay in the initialization.

Signed-off-by: marius-alex-tache <marius.tache@nxp.com>

Commit:
[NXP][platform][k32w] Small fixes for k32w platform code

 * correct typos
 * remove executable access permissions
 * remove unused files
 * change openthread_root to ot-nxp/openthread-k32w1

Signed-off-by: Andrei Menzopol <andrei.menzopol@nxp.com>

Commit:
[NXP][examples][k32w] Fixes for k32w examples code

 * small changes for existing code

Signed-off-by: Andrei Menzopol <andrei.menzopol@nxp.com>

dinabenamar committed:
[NXP][examples][common] Fix laundry-washer app by removing af-enums.h include in the operational-state-delegate-impl.h

marius-alex-tache committed:
[NXP][platform][k32w1] Fix gn issue

Signed-off-by: marius-alex-tache <marius.tache@nxp.com>

marius-alex-tache committed:
[NXP][platform][k32w1] Fix clang formatting

Signed-off-by: marius-alex-tache <marius.tache@nxp.com>

Commit:
[NXP][sdk][common] Remove chip_enable_icd_lit

 * argument has been moved to src/app/icd/icd.gni

Signed-off-by: Andrei Menzopol <andrei.menzopol@nxp.com>

Commit:
[NXP][examples][k32w][mw320] Remove not upstreamed and not supported apps

 * drop light-switch-combo-app as it isn't upstreamed yet
 * remove lock-app as it is not supported anymore
   (at least for the moment)
 * remove shell-app as it is not supported anymore
   (at least for the moment)

Signed-off-by: Andrei Menzopol <andrei.menzopol@nxp.com>

dinabenamar committed:
[NXP][examples][common] Updating laundry washer zap file to be consistent with current PICS

dinabenamar committed:
[NXP][examples][common] Updating thermostat zap files to be consistent with current PICS

doru91 committed:
[NXP][examples][k32w][contact-sensor][zap] Enable Diagnostic Logs Cluster

Signed-off-by: Doru Gucea <doru-cristian.gucea@nxp.com>

doru91 committed:
[NXP][platform][common] Add Diagnostic Logs common

Signed-off-by: Doru Gucea <doru-cristian.gucea@nxp.com>

doru91 committed:
[NXP][platform][common] create nxp_diagnostic_logs

Signed-off-by: Doru Gucea <doru-cristian.gucea@nxp.com>

doru91 committed:
[NXP][examples][k32w1] enable Diagnostic Logs for contact-sensor app

Signed-off-by: Doru Gucea <doru-cristian.gucea@nxp.com>

chapongatien committed:
[NXP][example][common] Fixing PLATFORM_InitTimerManager return value check issue

Signed-off-by: Gatien Chapon <gatien.chapon@nxp.com>

dinabenamar committed:
[NXP][examples][common] Fix laundry-washer-app build : remove zap_generated_dir from BUILD.gn as it does not exist anymore

doru91 committed:
[NXP][examples][common] Fix contact-sensor zap files

Signed-off-by: Doru Gucea <doru-cristian.gucea@nxp.com>

dinabenamar committed:
[NXP][doc] Adding support for the laundry-washer example inside docs/examples/index.md

dinabenamar committed:
[NXP][doc][rw61x] Update readme to support SDK 2.13.3

marius-alex-tache committed:
[NXP][scripts] Remove unsused binaries

Signed-off-by: marius-alex-tache <marius.tache@nxp.com>

marius-alex-tache committed:
[NXP][platform][k32w] Remove duplicated code introduced in a merge conflict

Signed-off-by: marius-alex-tache <marius.tache@nxp.com>

Commit:
[NXP][common] Update OT-NXP repo commit

Signed-off-by: Andrei Menzopol <andrei.menzopol@nxp.com>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/project-chip/connectedhomeip?pullRequest=32615) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/project-chip/connectedhomeip?pullRequest=32615) before we can accept your contribution.<br/>**5** out of **6** committers have signed the CLA.<br/><br/>:white_check_mark: marius-alex-tache<br/>:white_check_mark: dinabenamar<br/>:white_check_mark: marian-chereji-nxp<br/>:white_check_mark: andrei-menzopol<br/>:white_check_mark: doru91<br/>:x: chapongatien<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/project-chip/connectedhomeip?pullRequest=32615) it.</sub>

dinabenamar committed:
[NXP][CI][rw61x] Updating docker image version for SDK support

Commit:
[NXP][CI][k32w] Update workflows and docker image version for SDK support

Signed-off-by: Andrei Menzopol <andrei.menzopol@nxp.com>

Commit:
[NXP][examples][k32w] Fix spelling

Signed-off-by: Andrei Menzopol <andrei.menzopol@nxp.com>

dinabenamar committed:
[NXP][rw61x] Fix doc, spelling and zap errors

dinabenamar committed:
[NXP][rw61x] Fix doc spelling errors

marius-alex-tache committed:
[NXP][k32w0] Fix SDK path

Signed-off-by: marius-alex-tache <marius.tache@nxp.com>

restyled-commits committed:
Restyled by whitespace

restyled-commits committed:
Restyled by clang-format

restyled-commits committed:
Restyled by gn

restyled-commits committed:
Restyled by prettier-markdown

restyled-commits committed:
Restyled by autopep8

restyled-commits committed:
Restyled by isort

marius-alex-tache committed:
[NXP][doc][k32w] Fix docs issues

Signed-off-by: marius-alex-tache <marius.tache@nxp.com>

marius-alex-tache committed:
[NXP][CI][k32w] Remove deprecated examples

Signed-off-by: marius-alex-tache <marius.tache@nxp.com>

marius-alex-tache committed:
[NXP][docs] Remove lock app from commissioning guide

Signed-off-by: marius-alex-tache <marius.tache@nxp.com>

marius-alex-tache committed:
[NXP][platform][common] Remove unused header

Signed-off-by: marius-alex-tache <marius.tache@nxp.com>

marius-alex-tache committed:
[NXP][platform][k32w1] Remove custom openthread_root

Signed-off-by: marius-alex-tache <marius.tache@nxp.com>

dinabenamar committed:
[NXP][platform][rw61x] Fix lint code error

marius-alex-tache committed:
[NXP][k32w0] Fix gn import order

Signed-off-by: marius-alex-tache <marius.tache@nxp.com>

marius-alex-tache committed:
[NXP][sdk][k32w0] Fix SDK path condition

When official repo CI is used, the docker image sets the west github SDK
to /opt/sdk/core. This caused gn SDK files to mistakenly treat the SDK
as a package SDK, which affected the paths.

Signed-off-by: marius-alex-tache <marius.tache@nxp.com>

marius-alex-tache committed:
[NXP][scripts][k32w] Take into account low power flags for k32w0

Pigweed tokenizer logging and expansion board gn args are enabled by default
in the reference apps args.gni. Disable them explicitly for low power case.

Signed-off-by: marius-alex-tache <marius.tache@nxp.com>

marius-alex-tache committed:
[NXP][examples][k32w1] Move diagnostic_logs to a common examples area

This type of app specific implementation should reside in the application
space, instead of the device layer.

Signed-off-by: marius-alex-tache <marius.tache@nxp.com>

marius-alex-tache commented:
This should be the last PR with a lot of changes, since we've switched to a new dev process. The upcoming updates should avoid this kind of major changes and should be easier to review. The aim is to synchronize the upstream/downstream once, such that we can enable upcoming granular updates.

andy31415 reviewed (approved):
Looked at some common file changes. Since this is a 20K delta PR, I don't think it can be reviewed in detail in a reasonable timeframe.
